flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/sharedbuffer/SharedBuffer.java

```diff
@@ -75,6 +76,9 @@
 	private MapState<Long, Integer> eventsCount;
 	private MapState<NodeId, Lockable<SharedBufferNode>> entries;
 
+	private HashMap<EventId, Lockable<V>> eventsBufferCache = new HashMap<>();
+	private HashMap<NodeId, Lockable<SharedBufferNode>> entryCache = new HashMap<>();
```

Comment:
I would suggest change it to 

`private Map<EventId, Lockable<V>> eventsBufferCache = new HashMap<>();`

Reply:
agree and fixed